testsuite/tests/lib-dynarray/test.ml

```diff
@@ -529,6 +529,32 @@ let () =
   assert (A.length a = 201);
   assert (A.length a = A.capacity a);;
 
+(** unsafe_to_iarray *)
+
+let () =
+  let n = 42 in
+  let a = A.unsafe_to_iarray ~capacity:n (fun t ->
+    for i = 0 to n - 1 do
+      A.add_last t i
+    done)
+  in
+  for i = 0 to n - 1 do
+    assert (Iarray.get a i = i)
+  done;;
```

Comment:
Could there be tests for the situations where the capacity at the end is not `capacity`?